### ISSUE TITLE
sign_builder_transaction sometimes does not work

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#864]
Here is a simplified version of what I've encountered in production, perhaps unable to reproduce if my guess is wrong.

* account A is a normal account with an active key imported to cli_wallet;
* account B is a multi-sig account with:
  * owner: threshold 2, C weight 1, D weight 1, E weight 1;
  * active: threshold 2, C weight 1, D weight 1, E weight 1;
* The builder transaction is
  * proposal X
    * fee_paying_account is account A
    * operations:
      * transfer some BTS from B to F.

In this case, `sign_builder_transaction` command doesn't add signature correctly, but `sign_transaction` command with a json produced by `preview_builder_transaction` does work.

**Comment by pmconrad** [comment]
Confirmed.
```
unlocked >>> begin_builder_transaction 
begin_builder_transaction 
0
unlocked >>> add_operation_to_builder_transaction 0 [22,{"fee_paying_account":"1.2.20148","expiration_time":"2018-04-20T14:55:00","proposed_ops":[{"op":[0,{"from":"1.2.0","to":"1.2.20148","amount":{"amount":10000}}]}]}]
add_operation_to_builder_transaction 0 [22,{"fee_paying_account":"1.2.20148","expiration_time":"2018-04-20T14:55:00","proposed_ops":[{"op":[0,{"from":"1.2.0","to":"1.2.20148","amount":{"amount":10000}}]}]}]
null
unlocked >>> sign_builder_transaction 0 false
sign_builder_transaction 0 false
[...]
unlocked >>> sign_builder_transaction 0 true 
sign_builder_transaction 0 true
19690ms th_a       wallet.cpp:1872               sign_transaction     ] Caught exception while broadcasting tx 9647b74e488ae82188db9eeda22d1ef5859df4a7:  0 exception: unspecified                      
missing required active authority: Missing Active Authority 1.2.20148                               
```
Interestingly, if I leave out the ```sign_builder_transaction 0 false``` it works (in the sense that it complains about the missing fee not the missing authority).

I suspect the first invocation creates a signature that is later re-used although it is invalid.

**Comment by abitmore** [comment]
Fixed by #865
